src/ring_theory/valuation/valuation_ring.lean

```diff
@@ -270,6 +278,135 @@ end
 
 section
 
+variables {R : Type*} [comm_ring R] [is_domain R] {K : Type*}
+variables [field K] [algebra R K] [is_fraction_ring R K]
+
+lemma iff_dvd_total :
+  valuation_ring R ↔ is_total R (∣) :=
+begin
+  classical,
+  refine ⟨λ H, ⟨λ a b, _⟩, λ H, ⟨λ a b, _⟩⟩; resetI,
+  { obtain ⟨c,rfl|rfl⟩ := @@valuation_ring.cond _ _ H a b; simp },
+  { obtain (⟨c, rfl⟩|⟨c, rfl⟩) := @is_total.total _ _ H a b; use c; simp }
+end
+
+lemma iff_ideal_total :
+  valuation_ring R ↔ is_total (ideal R) (≤) :=
+begin
+  classical,
+  refine ⟨λ _, by exactI ⟨le_total⟩, λ H, iff_dvd_total.mpr ⟨λ a b, _⟩⟩,
+  have := @is_total.total _ _ H (ideal.span {a}) (ideal.span {b}),
+  simp_rw ideal.span_singleton_le_span_singleton at this,
+  exact this.symm
+end
+
+variable {R}
+
+lemma dvd_total [h : valuation_ring R] (x y : R) : x ∣ y ∨ y ∣ x :=
+@@is_total.total _ (iff_dvd_total.mp h) x y
+
+lemma unique_irreducible [valuation_ring R] ⦃p q : R⦄
+  (hp : irreducible p) (hq : irreducible q) : associated p q :=
+begin
+  have := dvd_total p q,
+  rw [irreducible.dvd_comm hp hq, or_self] at this,
+  exact associated_of_dvd_dvd (irreducible.dvd_symm hq hp this) this,
+end
+
+lemma iff_is_integer_or_is_integer :
```

Comment:
Similarly, `K` should be explicit here.

Reply:
I have changed `R` and `K` to be explicit in this lemma since `K` doesn't appear on the LHS.